### PR TITLE
Initial prototype of using mermaid to make an interactive simulator

### PR DESCRIPTION
**Pull request by emmby**
Proof of concept for generating interactive mermaid diagrams from state machines.

Supports
- buttons for triggers
- displaying the currently active state
- codegen

### CODE REVIEW

**Comment by adamfk** [comment]
Very cool! Is the mermaid file currently handwritten?

**Comment by emmby** [comment]
sorry, I said it was but I lied, that part's still WIP

**Comment by adamfk** [comment]
> sorry, I said it was generated but I lied, that part's still WIP

No worries. Just wanted to make sure I didn't miss something.

**Comment by emmby** [comment]
Check out LightSm.html for an example of an interactive mermaid diagram

**Comment by adamfk** [comment]
> Check out LightSm.html for an example of an interactive mermaid diagram. I think it's going to be fairly simple to generate, will work on that next.

It might be useful to use the Visitor Pattern while generating the mermaid diagram text. Here's an example of the visitor pattern in state smith: https://github.com/StateSmith/StateSmith-examples/tree/main/visitor-graph-1

**Comment by emmby** [comment]
Updated to include codegen.
`dotnet-script LightSm.csx` will generate LightSm.js and LightSm.html.

File: test-gen-2/src/LightSm.h
Comment (adamfk) [inline test-gen-2/src/LightSm.h:1]: should this file be removed?

File: test-gen-2/src/LightSm.csx [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+SmRunner htmlRunner = new(diagramPath: "LightSm.drawio.svg", new LightSmRenderConfig(), transpilerId: TranspilerId.JavaScript);
+htmlRunner.SmTransformer.InsertBeforeFirstMatch(
+    StandardSmTransformer.TransformationId.Standard_FinalValidation,
+    new TransformationStep(id: "some string id", action: (sm) =>
+    {
+        var visitor = new MermaidGenerator(mermaidCodeWriter);
+        sm.Accept(visitor);
+        visitor.Print(); // print the mermaid code to the mermaidcodewriter
+        List<string> events = GetNonDoEvents(sm);
+        events = events.ConvertAll(e => e.ToUpper()); // TODO what is the right way to get the event name in the proper case?
```
Comment (adamfk) [inline test-gen-2/src/LightSm.csx:23]: We can easily get the events from the js state machine: https://github.com/StateSmith/exp-test-gen/blob/6b4fe3a9d69ef599280a7af400d9cec06a7f7504/src2/LightSm2Sim/index.html#L12-L21

Comment (emmby) [reply test-gen-2/src/LightSm.csx]: okay that is 1000x better

**Pull request merged**